### DISCUSSION TITLE
Copy widget preset (like clone_from)

### DISCUSSION THREAD

**Discussion by d47081** [gtk-rs/gtk4-rs#1932] [Q&A] [answered]
I want to copy all [TextTag](https://docs.gtk.org/gtk4/class.TextTag.html) settings to new one, something like clone, but for new object in memory. For example I have some link, with foreground, underline and other options preset and want to grab these defaults.

Is it possible with `gtk4-rs` methods or I should manually implement that for every member I want to grab?

**Comment by bilelmoussaoui** [comment, marked as answer]
Objects might have internals that cannot be cloned externally. So such methods cannot be automatically provided.

You will have to either write such code for your specific use case in the application side or edit glib to add such Clone interface for GObjects so that Rust can use.

**Reply by d47081** [reply]
Thank you for quick reply!
